### DISCUSSION TITLE
Maximum records in a query

### DISCUSSION THREAD

**Discussion by titobcn** [rappen/FetchXMLBuilder#879] [General]
Hello,
I have a query with more than 2000 records, when I want to save the view it doesn't let me and I get the error "Number of conditions in query exceeded maximum limit."
What can I do to be able to save the view with all the records that I have made through the fetch?
Thank you

**Comment by rappen** [comment]
I'm curious - can you actually execute the query in FXB? And it works?
Error message, will it stop you from the server? Screenshot?

**Reply by titobcn** [reply]
hello,
I have to perform a query with 1,000 records.
Attached image
![fetch errro](https://user-images.githubusercontent.com/127776774/224997569-1a23817c-0ffc-4512-812d-f274e5964f5f.GIF)

what i want is
Create a view with the emails of specific contacts
With this view, inform me of the id
Thanks to this id, I make sure that with an export to the online office I update the contacts with a special segmentation

Thanks!!

**Reply by rappen** [reply]
Ok, so the error message is not from our tool, it's from the Dataverse.
Is it not possible to create the query with joins of the filtering?
Generating a query with a bunch of guids is not so secure, since you don't really know when it will hit the limit of number of conditions.

**Reply by titobcn** [reply]
Wow...so what can I do?

**Reply by rappen** [reply]
Can you share your queries? If too sensitive to share publicly, can you email me?
Just to get an idea of what we need.

**Reply by titobcn** [reply]
You can provide an email
Thanks!

**Reply by rappen** [reply]
jonas at jonasr dot app

**Comment by titobcn** [comment]
Have you been able to see the file?

**Reply by rappen** [reply]
Yes.
This filtering is not possible with that many filters. Can't do.
I'm thinking about how to find all these emailaddresses?
I understand through Office, you should probably set a flag in these contacts with [Power Automate](https://learn.microsoft.com/en-us/training/powerplatform/power-automate?WT.mc_id=BA-MVP-5002475), and make the filtering a lot easier in the query:
```
<fetch>
  <entity name="contact">
    <!-- attributes -->
    <order attribute="ab_aplicacionorigen" descending="false" />
    <filter type="and">
      <condition attribute="statecode" operator="eq" value="0" />
      <condition attribute="ab_aplicacionorigen" operator="eq" value="{8437AA9C-D384-E811-80D9-005056BD6621}" />
      <condition attribute="ab_officesegment" operator="eq" value="true" />
    </filter>
  </entity>
</fetch>
```
(see the last condition above)

**Comment by titobcn** [comment]
Hello,
At the end I am going to export the contacts, update and import again.
It is the only feasible solution to unravel the project.
We are limited with the dataverse.
Thanks for the help!

**Reply by rappen** [reply]
Thanks and good luck! 🤓
